### ISSUE TITLE
Strict input parsing of realization configuration

### ISSUE THREAD

**Issue by PhilMiller** [NOAA-OWP/ngen#573]
It should be possible to have the code warn or error out on encountering an unrecognized key in the input configuration. This would enable better detection of typos in names of optional keys that might otherwise go unnoticed, among other possible mistaken inputs.

## Current behavior

Per #570, 
> Notably, existing formulations that have this key in their JSON (at least as false) will not break, the key will just be ignored

## Expected behavior

When enabled, the code should not ignore keys it doesn't understand

**Comment by mattw-nws** [comment]
Yes, we've bandied about the idea of leveraging the work in [`ngen_conf`](https://github.com/NOAA-OWP/ngen-cal/tree/master/python/ngen_conf) and doing something like this via pybind11... which seems to make some sense.

As long as we still don't have a JSON parser that supports comments however, I strongly recommend making an unrecognized key a warning rather than a failure--this is about the only way to get ngen to ignore something in the file at the moment without comment support.
